lib/unicode.ml

```diff
@@ -126,26 +126,29 @@ let utf8_of_unicode n =
   if n < 128 then
     String.make 1 (Char.chr n)
   else if n < 2048 then
-    let s = String.make 2 (Char.chr (128 + n mod 64)) in
-    begin
-      s.[0] <- Char.chr (192 + n / 64);
-      s
-    end
+    String.init 2 (fun idx ->
+        match idx with
+        | 0 -> Char.chr (192 + n / 64)
+        | 1 -> Char.chr (128 + n mod 64)
+        | _ -> 'x'
+      )
   else if n < 65536 then
-    let s = String.make 3 (Char.chr (128 + n mod 64)) in
-    begin
-      s.[1] <- Char.chr (128 + (n / 64) mod 64);
-      s.[0] <- Char.chr (224 + n / 4096);
-      s
-    end
+    String.init 3 (fun idx ->
+        match idx with
+        | 0 -> Char.chr (224 + n / 4096)
+        | 1 -> Char.chr (128 + (n / 64) mod 64)
+        | 2 -> Char.chr (128 + n mod 64)
+        | _ -> 'x'
```

Comment:
I'm unsure there is a rule allowing to say that depending on an external library is necessarily better. For instance, camomile on which we hesitated to rely a few years ago did not evolve for more than 5 years and their unicode tables are not any more up-to-date.